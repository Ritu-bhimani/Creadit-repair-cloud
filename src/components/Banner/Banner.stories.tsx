import { Typography } from '@mui/material';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';
import ReactPlayer from 'react-player';
import { Banner } from './Banner';

export default {
    title: 'Banner',
    component: Banner
} as ComponentMeta<typeof Banner>;

const Template: ComponentStory<typeof Banner> = (args) => (
    <Banner {...args} />
);

export const Primary = Template.bind({});
const VideoContent = () => (
    <React.Fragment>
        <h2></h2>
        <ReactPlayer url="https://player.vimeo.com/video/560100304?badge=0&player_id=0?autoplay=0&amp;modestbranding=1&amp;showinfo=0" width="100%" controls />
    </React.Fragment>
);
const Children = () => {
    return (
        <Typography
            variant="h4"
            gutterBottom
            className="font-26-400"
            style={{ marginBottom: '8px' }}
        >
            Welcome To Credit Repair Cloud!
        </Typography>
    )
}
Primary.args = {
    videoLength: '15.30',
    overLayImage: 'https://qa.creditrepaircloud.com/assets/images/banners/Home.png',
    children: <Children />,
    videoTitle: 'Welcome To Credit Repair Cloud!',
    videoContent: <VideoContent />,

};
