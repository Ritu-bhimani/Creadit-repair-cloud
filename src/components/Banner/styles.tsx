import { css } from '@emotion/react';

export const BannerStyles = {
  heading: css`
    margin-top: 40px;
    margin-bottom: 1.2em;
    line-height:25px;
  `,
  bannerContainer: css`
   &:hover {
      box-shadow: 0 0 4px 1px #78b3efd9 !important;
    }
  `,
  bannerChildrenContainer: css`
    position: relative;
    justify-content: space-between;
    background-color: #cbe5ff;
    padding: 23px 35px 35px 35px;
    height:100%;
    a {
      color: #0075cc !important;
      cursor: pointer;
      text-decoration: none;
      background-color: transparent;
    }
    a:hover {
      text-decoration: underline;
      color: #244894 !important;
    }`,
  videoContainer: css`
    cursor: pointer;
    width: 324px;
    height: 198px;
    position: relative;
    &:hover .vidioTitle{
    text-decoration: underline!important;
   }
    `,
  video: css`
   &:hover {
    filter: contrast(0.5)!important;
   }
  
  `,
  image: css`
    width:100%;
    height: 164px;
    vertical-align: middle;
    border-style: none;
  `,
  playCircleIcon: css`
    position: absolute;
    top: 36%;
    left: 43%;

  `,
  textWhite: css`
    color: #fff !important;

  `,
  videoDetails: css`
    padding-top: 9px;
    padding-bottom: 9px;
    padding-left: 16px;
    padding-right: 16px;
    --tw-bg-opacity: 1;
    background-color: rgb(102 102 102 / var(--tw-bg-opacity));
    justify-content: space-between;
    align-items: center;
    font-weight: 600;
    font-size: 0.75rem;
    line-height: 1rem;
    display: flex;
    color:  #fff !important;
  `,
  vidioTitle: css`
    color: #fff !important;
  `,
  quickVideo: css`
    color: #0075cc!important;
    cursor: pointer;
    text-decoration: none;
    background-color: transparent;
    margin-left: 12px;
    margin-top: 10px;
    .quick-video {
      font-size: 14px;
      vertical-align: top;
    }
    &:hover {
      color:#244894!important;
    }
    .quick-video {
    &:hover {
      text-decoration: underline;

    }`,
  dismissLink: css`
    position: absolute;
    bottom:20px;
    font-weight: 600 !important;
    `

};