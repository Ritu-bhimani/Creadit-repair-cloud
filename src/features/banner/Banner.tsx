import { Box, CircularProgress, css, Grid, Link, Typography } from '@mui/material';
import { FC, useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import ReactPlayer from 'react-player';
import { Banner } from '../../components';


const bannerTitle = css`
  font-weight: 400 !important;
  font-size: 26px !important;
  color: #4a4a4a;
`;
const bannerPara = css`
  font-size: 14px !important;
  font-weight: 100 !important;
  color: #4a4a4a;
  line-height: 1.7;
`;

interface HomeBannerProps {
  pageTitle: string;
}
export const HomeBanner: FC<HomeBannerProps> = (props: HomeBannerProps) => {
  const { pageTitle } = props;
  const { t } = useTranslation();
  const CrcSoftwareDemoUrl = "https://player.vimeo.com/video/560100304?badge=0&player_id=0?autoplay=0&amp;modestbranding=1&amp;showinfo=0";
  const CrcWelcomeUrl = "https://youtu.be/38CxxH_HRbA";
  const CrcVideoThumbnailUrl = "https://qa.creditrepaircloud.com/assets/images/banners/Home.png";
  const SampleClientUrl = "#";
  const UserGuideUrl = "https://www.creditrepaircloud.com/resources";
  const ClickHereUrl = "#";
  const [modelTitle, setModelTitle] = useState(t('creditRepairCloudSoftwareDemo'));
  const [videoUrl, setVideoUrl] = useState(CrcSoftwareDemoUrl);
  const [showModel, setShowModel] = useState(false)
  const [videoContent, SetVideoContent] = useState(<></>)
  const [isLoading, setIsLoading] = useState(true)

  const handleRedirection = (value: string) => {
    if (value === 'welcome') {
      setVideoUrl(CrcWelcomeUrl)
      setModelTitle(t('videoPreview'))
      setIsLoading(true)
    } else if (value === 'tour') {
      setVideoUrl("")
      setModelTitle(t('danielFromCreditRepairCloud'))
      setIsLoading(false)
    }
    setShowModel(true)


  }
  useEffect(() => {
    if (showModel === false) {
      setVideoUrl(CrcSoftwareDemoUrl)
      setModelTitle(t('creditRepairCloudSoftwareDemo'))
      setIsLoading(true)
    }
    const VideoContent = (
      <>{isLoading &&
        <Box sx={{position:'absolute', left:'47%', top:'47%'}}>
          <CircularProgress />
        </Box>
      }

        <ReactPlayer
          url={videoUrl}
          width="100%"
          controls
          onReady={() => setIsLoading(false)}
        />
      </>
    );
    SetVideoContent(VideoContent)
  }, [showModel, isLoading])
  const BannerChildren = () => {
    return (
      <Grid
        container
        className="bannerDesc"
        direction="column"
      >
        <div>
          <Typography variant="h5" gutterBottom css={bannerTitle}>
            {t('bannerWelcome')}
          </Typography>
          <Typography gutterBottom css={bannerPara}>
            {t('bannerText')}
            <Link onClick={() => handleRedirection('welcome')}>{t('bannerVideo')}</Link>
            {t('bannerThe')}
            <Link onClick={() => handleRedirection('tour')}>{t('bannerTour')}</Link>
            {t('bannerLearn')}
            <Link href={SampleClientUrl}>{t('bannerClient')}</Link>
            {t('bannerReadOur')}
            <Link
              href={UserGuideUrl}
              target="_blank"
              rel="noreferrer"
            >
              {t('bannerUserGuide')}
            </Link>
            {t('bannerChanges')}
            <Link href={ClickHereUrl}>{t('bannerClickHere')}</Link>
          </Typography>
        </div>
      </Grid>
    );
  };
  return (
    <Banner
      videoLength="15:45"
      modelTitle={modelTitle}
      overLayImage={CrcVideoThumbnailUrl}
      videoTitle={t('bannerWelcome')}
      videoContent={videoContent}
      setShowModal={setShowModel}
      showModel={showModel}
      pageTitle={pageTitle}
    >
      <BannerChildren />
    </Banner>

  );
};
