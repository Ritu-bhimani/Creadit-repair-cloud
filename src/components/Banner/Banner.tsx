import { PlayCircle } from '@mui/icons-material';
import PlayCircleIcon from '@mui/icons-material/PlayCircle';
import { Typography } from '@mui/material';
import Grid from '@mui/material/Grid';
import React, { FC, useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Modal } from '../Modal';
import { BannerStyles } from './styles';

interface BannerProp {
  videoLength?: string;
  overLayImage?: string;
  children?: React.ReactNode;
  videoId?: string;
  videoTitle?: string | null;
  videoUrl?: string;
  videoContent?: React.ReactNode;
  modelTitle?: string | null;
  showModel?: any;
  setShowModal?: any;
  pageTitle?: string | null;
}
export const Banner: FC<BannerProp> = (props: BannerProp) => {
  const { t } = useTranslation();
  const [showBanner, setShowBanner] = useState(false);

  const bannerStatus = localStorage.getItem('banner');
  const disableBanner = () => {
    setShowBanner(false);
    localStorage.setItem('banner', 'Disabled');
  };
  useEffect(() => {
    if (props.children === undefined) {
      setShowBanner(false);
    }
  });
  const handleBanner = () => {
    if (props.children === undefined) {
      setShowModal(true);
    } else {
      setShowBanner(true);
    }
  };
  useEffect(() => {
    if (bannerStatus) {
      setShowBanner(false);
    } else {
      setShowBanner(true);
    }
  }, [bannerStatus]);
  const {
    videoLength,
    overLayImage,
    children,
    videoTitle,
    videoContent,
    modelTitle,
    showModel,
    setShowModal,
    pageTitle
  } = props;

  return (
    <>
      <Modal
        closeIconVisible
        width="sm"
        open={showModel}
        title={modelTitle}
        closeOnEscape={true}
        onClose={() => {
          setShowModal(false);
        }}
      >
        {videoContent}
      </Modal>

      {showBanner && props.children !== undefined && (
        <Grid
          container
          style={{ margin: '20px 0px 20px 0px' }}
          css={BannerStyles.bannerContainer}
        >
          <Grid
            item
            xs={12}
            md={3}
            css={BannerStyles.videoContainer}
            onClick={() => setShowModal(true)}
          >
            <div css={BannerStyles.video}>
              <img src={overLayImage} css={BannerStyles.image}></img>
            </div>
            <div css={BannerStyles.playCircleIcon}>
              <PlayCircleIcon
                css={BannerStyles.textWhite}
                sx={{ fontSize: '54px' }}
              />
            </div>
            <div css={BannerStyles.videoDetails}>
              <span css={BannerStyles.vidioTitle} className="vidioTitle">
                {videoTitle}
              </span>
              <span>{videoLength}</span>
            </div>
          </Grid>
          <Grid item xs={12} md={9}>
            <Grid css={BannerStyles.bannerChildrenContainer}>
              {children}

              <div css={BannerStyles.dismissLink}>
                <a onClick={() => disableBanner()}>{t('bannerDismiss')}</a>
              </div>
            </Grid>
          </Grid>
        </Grid>
      )}
      <Typography component={'div'} variant="h2" css={BannerStyles.heading}>
        {pageTitle}
        {!showBanner && (
          <a css={BannerStyles.quickVideo} onClick={() => handleBanner()}>
            <PlayCircle style={{ marginRight: '5px' }} />
            <span className="quick-video">Quick Video</span>
          </a>
        )}
      </Typography>
    </>
  );
};

export default Banner;
