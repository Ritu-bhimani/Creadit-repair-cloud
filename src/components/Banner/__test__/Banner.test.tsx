import { render } from '@testing-library/react';
import { Banner } from '../Banner';
it('loads and displays banner', () => {
  const component = render(<Banner showModel={true} setShowModal={true} />);
  expect(component).toMatchSnapshot();
});
