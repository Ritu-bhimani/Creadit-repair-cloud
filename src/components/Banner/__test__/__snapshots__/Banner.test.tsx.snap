// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays banner 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        style="margin: 40px 0px 40px 0px;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <img
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            />
          </div>
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1g17bvt-MuiSvgIcon-root"
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
              data-testid="PlayCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.5 16.5v-9l7 4.5-7 4.5z"
              />
            </svg>
          </div>
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <span
              class="vidioTitle"
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            />
            <span />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
        >
          <div
            class="MuiGrid-root css-vj1n65-MuiGrid-root"
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <div
              css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            >
              <a>
                bannerDismiss
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      />
    </div>
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            <div
              class="MuiBox-root css-70qvj9"
            >
              <div
                class="MuiBox-root css-i9gxme"
              />
              <div
                class="MuiBox-root css-0"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-fpx798-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          />
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      style="margin: 40px 0px 40px 0px;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <img
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          />
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1g17bvt-MuiSvgIcon-root"
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
            data-testid="PlayCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM9.5 16.5v-9l7 4.5-7 4.5z"
            />
          </svg>
        </div>
        <div
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <span
            class="vidioTitle"
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          />
          <span />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          <div
            css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          >
            <a>
              bannerDismiss
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
